lib/Macros/Sources/ObservationMacros/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+  )
+
+  target_compile_options(ObservationMacros PRIVATE
+    $<$<COMPILE_LANGUAGE:Swift>:-runtime-compatibility-version>
+    $<$<COMPILE_LANGUAGE:Swift>:none>)
+
+  # Set the appropriate target triple.
+  if(SWIFT_HOST_VARIANT_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
+    set(DEPLOYMENT_VERSION "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_DEPLOYMENT_VERSION}")
+  endif()
+
+  if(SWIFT_HOST_VARIANT_SDK STREQUAL ANDROID)
+    set(DEPLOYMENT_VERSION ${SWIFT_ANDROID_API_LEVEL})
+  endif()
+
+  get_target_triple(target target_variant "${SWIFT_HOST_VARIANT_SDK}" "${SWIFT_HOST_VARIANT_ARCH}"
+    MACCATALYST_BUILD_FLAVOR ""
+    DEPLOYMENT_VERSION "${DEPLOYMENT_VERSION}")
+
+    target_compile_options(ObservationMacros PRIVATE $<$<COMPILE_LANGUAGE:Swift>:-target;${target}>)
+ 
+  # Workaround a cmake bug, see the corresponding function in swift-syntax
+  function(force_target_macros_link_libraries TARGET)
+    cmake_parse_arguments(ARGS "" "" "PUBLIC" ${ARGN})
+
+    foreach(DEPENDENCY ${ARGS_PUBLIC})
+      target_link_libraries(${TARGET} PRIVATE
+        ${DEPENDENCY}
+      )
+      add_dependencies(${TARGET} ${DEPENDENCY})
+
+      add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+        COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+        DEPENDS ${DEPENDENCY}
+        )
+      target_sources(${TARGET} PRIVATE
+        ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+      )
+    endforeach()
+  endfunction()
+
+  set(SWIFT_SYNTAX_MODULES
+    SwiftSyntax
+    SwiftSyntaxMacros
+  )
+
+  # Compute the list of SwiftSyntax targets
+  list(TRANSFORM SWIFT_SYNTAX_MODULES PREPEND "SwiftSyntax::"
+       OUTPUT_VARIABLE SWIFT_SYNTAX_TARGETS)
+
+  # TODO: Change to target_link_libraries when cmake is fixed
+  force_target_link_libraries(ObservationMacros PUBLIC
+    ${SWIFT_SYNTAX_TARGETS}
+  )
+
+  set(SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR
+      "${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/lib/swift/host")
+  set(SWIFT_SYNTAX_LIBRARIES_DEST_DIR
+      "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/swift/host")
+
+  # Determine the SwiftSyntax shared library files that were built as
+  # part of earlyswiftsyntax.
+  list(TRANSFORM SWIFT_SYNTAX_MODULES PREPEND ${CMAKE_SHARED_LIBRARY_PREFIX}
+       OUTPUT_VARIABLE SWIFT_SYNTAX_SHARED_LIBRARIES)
+  list(TRANSFORM SWIFT_SYNTAX_SHARED_LIBRARIES APPEND
+       ${CMAKE_SHARED_LIBRARY_SUFFIX}
+       OUTPUT_VARIABLE SWIFT_SYNTAX_SHARED_LIBRARIES)
+
+  # Copy over all of the shared libraries from earlyswiftsyntax so they can
+  # be found via RPATH.
+  foreach (sharedlib ${SWIFT_SYNTAX_SHARED_LIBRARIES})
+    add_custom_command(
+      TARGET ObservationMacros PRE_BUILD
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR}/${sharedlib} ${SWIFT_SYNTAX_LIBRARIES_DEST_DIR}/${sharedlib}
+      COMMENT "Copying ${sharedlib}"
+    )
+  endforeach()
+
```

Comment:
I don't understand why we need this. ASTGen should already have dealt with whatever is needed?

Reply:
most of this was copy-pasta from ASTGen